### PR TITLE
Update doc about IDE file link format

### PR DESCRIPTION
**Pull request by nicolas-grekas**
todo:
- [x] ~~tell about `kernel.file_link_format` parameter?~~
- [x] tell about prefix maps (see symfony/symfony#19807)

### CODE REVIEW

File: reference/configuration/framework.rst [hunk 1]
```diff
@@ -204,21 +204,20 @@ ide
...
-* ``textmate``
-* ``macvim``
-* ``emacs``
-* ``sublime``
+Alternatively, you can use this ``ide`` configuration key.
 
-You can also specify a custom URL string. If you do this, all percentage
-signs (``%``) must be doubled to escape that character. For example, if
-you use PHPstorm on the Mac OS platform, you will do something like:
+In both cases, the expected configuration value is a URL template that should
```
Comment (javiereguiluz) [inline reference/configuration/framework.rst]: `that should contain` -> `that contains`

**Comment by javiereguiluz** [comment]
👍 

Status: reviewed

File: reference/configuration/framework.rst [hunk 2]
```diff
@@ -204,21 +204,20 @@ ide
 
 **type**: ``string`` **default**: ``null``
 
-If you're using an IDE like TextMate or Mac Vim, then Symfony can turn all
-of the file paths in an exception message into a link, which will open that
-file in your IDE.
+Symfony can turn file paths seen in dumps and in exception messages into links
```
Comment (wouterj) [inline reference/configuration/framework.rst]: you can remove the "in" before "exception messages", which will improve the read flow a bit

File: reference/configuration/framework.rst [hunk 3]
```diff
@@ -204,21 +204,20 @@ ide
...
-* ``textmate``
-* ``macvim``
-* ``emacs``
-* ``sublime``
+Alternatively, you can use this ``ide`` configuration key.
 
-You can also specify a custom URL string. If you do this, all percentage
-signs (``%``) must be doubled to escape that character. For example, if
-you use PHPstorm on the Mac OS platform, you will do something like:
+In both cases, the expected configuration value is a URL template that contain an
```
Comment (wouterj) [inline reference/configuration/framework.rst]: "that contain an" -> "that contain**s** an"

File: reference/configuration/framework.rst [hunk 4]
```diff
@@ -204,21 +204,20 @@ ide
...
+``%f`` where the file path is expected and ``%l`` for the line. When using the
```
Comment (wouterj) [inline reference/configuration/framework.rst]: "for the line" -> "for the line numbers" (or is it the actual line?)

File: reference/configuration/framework.rst [hunk 5]
```diff
@@ -248,18 +247,25 @@ you use PHPstorm on the Mac OS platform, you will do something like:
...
 
+.. versionadded:: 3.2
+
+    When running your app in a container or in a virtual machine, you can tell
+    Symfony to map files from the guest to the host by changing their prefix.
+    This map should be specified at the end of the URL template::
+
+        // /guest/.../file will be opened as /host/.../file on the host
+        // and /foo/.../file as /bar/.../file also
+        'myide://%f:%l#/guest/=/host/&/foo/=/bar/&...'
```
Comment (wouterj) [inline reference/configuration/framework.rst]: please move this out the of the versionadded and into a tip/note (or just normal text). The reason is that we regularly remove the versionadded directives in newer versions of the docs, which would mean loosing this nice information atm.

Comment (wouterj) [inline reference/configuration/framework.rst]: I think we should explain the format a bit more. Afaics, you have to differentiate the URL template and the prefixes with the `#` character? (should be explained)

Also, `/guest/` seem to indicate that's a regex?

**Comment by wouterj** [comment]
status: needs work

**Comment by nicolas-grekas** [comment]
comments addressed

**Review by wouterj — APPROVED** [review]

**Comment by wouterj** [comment]
👍 
status: reviewed

looks good now

**Comment by xabbuh** [comment]
Thank you Nicolas.

**Pull request merged**